src/monitors/turbFlowStats.cpp

```diff
@@ -26,253 +21,172 @@ PetscErrorCode ablate::monitors::TurbFlowStats::MonitorTurbFlowStats(TS ts, Pets
+        for (std::size_t f = 0; f < monitor->fieldNames.size(); f++) {
+            const auto& field = monitor->GetSolver()->GetSubDomain().GetField(monitor->fieldNames[f]);
+            ierr = monitor->GetSolver()->GetSubDomain().RestoreFieldGlobalVector(field, &vecIS[f], &vec[f], &fieldDM[f]);
+            CHKERRQ(ierr);
+        }
     }
     PetscFunctionReturn(0);
 }
 
-void ablate::monitors::TurbFlowStats::AddField(DM& dm, const char* nameField, PetscInt numComp) {
-    PetscFV fvm;
-    PetscFVCreate(PetscObjectComm(PetscObject(dm)), &fvm) >> checkError;
-    PetscObjectSetName((PetscObject)fvm, nameField) >> checkError;
-    PetscFVSetFromOptions(fvm) >> checkError;
-    PetscFVSetNumComponents(fvm, numComp) >> checkError;
-
-    DMAddField(dm, nullptr, (PetscObject)fvm) >> checkError;
-    PetscFVDestroy(&fvm);
-}
-
 void ablate::monitors::TurbFlowStats::Register(std::shared_ptr<ablate::solver::Solver> solverIn) {
-    // Copy over the solver
-    ablate::monitors::Monitor::Register(solverIn);
-
-    // Copy the master DM over, set fields, and initialize vector values
-    this->GetSolver()->GetSubDomain().CreateEmptySubDM(&turbDM);
-    std::string DmName = "TurbDM";
-    PetscObjectSetName((PetscObject)turbDM, DmName.c_str());
-
-    // Add the densitySum and densityDtSum fields
-    std::string densitySum = "rhoSum";
-    AddField(turbDM, densitySum.c_str(), 1);
+    // Create the monitor name
+    std::string dmID = "TurbFlowStats";
 
-    std::string densityDtSum = "rhoDtSum";
-    AddField(turbDM, densityDtSum.c_str(), 1);
+    // Create suffix vector
+    std::vector<std::string> suffix{"_rhoMult", "_rhoDtMult", "_rhoSqr", "_sum", "_sumSqr", "_favreAvg", "_rms", "_mRms"};
 
-    // Initialize the numComps variable, which will store the sum of all the fields
-    PetscInt numComp = 0;
-
-    // Resize the fieldTrack and fieldComps vectors to be the proper size (fieldTrack one greater than num fields, fieldComps equal to num fields)
-    fieldComps.resize(fieldNames.size(), 0);
-    fieldTrack.resize(fieldNames.size() + 1, 0);
-    fieldTrack[0] = 0;
-
-    // Need to find the total number of field components and the indices marking the beginning of each field
+    // Create vectors of names for all components of all fields
+    std::vector<std::vector<std::string>> processedCompNames;
     for (std::size_t f = 0; f < fieldNames.size(); f++) {
-        const auto& field = this->GetSolver()->GetSubDomain().GetField(fieldNames[f]);
-
-        fieldComps[f] = field.numberComponents;
-        numComp += field.numberComponents;
-        if ((f + 1) < fieldTrack.size()) {
-            fieldTrack[f + 1] = fieldTrack[f] + field.numberComponents;
+        const auto& field = solverIn->GetSubDomain().GetField(fieldNames[f]);
+        std::vector<std::string> innerCompNames(8 * field.numberComponents);
+        for (PetscInt c = 0; c < field.numberComponents; c++) {
+            for (std::size_t p = 0; p < suffix.size(); p++) {
```

Comment:
Would you mind removing the hard code 8 and 2.  If these are based upon the enums in the header file you can do something like https://stackoverflow.com/questions/2102582/how-can-i-count-the-items-in-an-enum